src/basic/os-util.c

```diff
@@ -460,7 +460,7 @@ int os_release_support_ended(const char *support_end, bool quiet, usec_t *ret_eo
                                       "Failed to parse SUPPORT_END= in os-release file, ignoring: %m");
 
         time_t eol = timegm(&tm);
-        if (eol == (time_t) -1)
+        if (eol <= (time_t) -1)
                 return log_full_errno(quiet ? LOG_DEBUG : LOG_WARNING, SYNTHETIC_ERRNO(EINVAL),
                                       "Failed to convert SUPPORT_END= in os-release file, ignoring: %m");
```

Comment:
Please just check < 0 then. Plus, though not your fault, `%m` should not be used alongside `SYNTHETIC_ERRNO`. Hence the log message should be adjusted to be e.g. "Parsed SUPPORT_END= time out of range, refusing"